spec/abci/abci++_app_requirements.md

```diff
@@ -188,6 +188,37 @@ Likewise, `ExtendVote` can also be non-deterministic:
 * *w<sup>r</sup><sub>p</sub> = w<sup>r</sup><sub>q</sub> &#8655;
   e<sup>r</sup><sub>p</sub> = e<sup>r</sup><sub>q</sub>*
 
+Unlike all requirements described above,
+which affect the ABCI methods of the [Consensus Connection](#consensus-connection),
+the following requirement is on the [Mempool Connection](#mempool-connection).
+Let *CheckTxCode<sub>tx,p,t</sub>* denote the error code returned by *p*'s Application, via `ResponseCheckTx`,
+to CometBFT's `RequestCheckTx` call occurring at time *t* and having transaction *tx* as parameter.
+Let predicate *OK(CheckTxCode)* denote whether *CheckTxCode* is `SUCCESS`.
+
+
+* Requirement 13 [`CheckTx`, finite-oscillation]: For any correct process *p*
+  and any transaction *tx*,
+  
+    - (a) either *OK(CheckTxCode<sub>tx,p,t<sub>0</sub></sub>) = OK(CheckTxCode<sub>tx,p,t<sub>1</sub></sub>)*
+      for any two times *t<sub>0</sub>* and *t<sub>1</sub>* in *p*'s execution, or
+    - (b) there exists a time *t<sub>stable</sub>* in *p*'s execution
+      such that *&#172; OK(CheckTxCode<sub>tx,p,t</sub>)* for all *t > t<sub>stable</sub>*.
+
+Requirement 13 has two main conditions. The implementation of `CheckTx` must fulfill at least one of them.
+Condition (a) mandates that `CheckTx` on a transaction never changes its result,
+in other words, that the checks implemented are stateless (e.g., well-formedness, signature verification).
+Condition (b) states that, if the transaction stays in the mempool long enough, it will eventually fail
+`CheckTx` forever.
+Condition (b) is aiming at `CheckTx` implementations that read the Application's state,
+whose result can oscillate as the tx might be validated against different states in successive calls to `CheckTx`.
+Thus, condition (b), while being local to a process *p*, will make sure that the transaction will eventually
+leave the mempool of *all* full nodes.
```

Comment:
Is that including the possibility of `tx` been included in a committed block? This should be, in my opinion, provided as the main example of when/how a transaction becomes "forever" invalid.

Reply:
The PR is trying to provide requirements to the mempool without having to link it to consensus. If that's possible, I prefer it that way (more modular)